eng/Versions.props

```diff
@@ -28,7 +28,7 @@
   <PropertyGroup>
     <!-- Versions used by several individual references below -->
     <RoslynDiagnosticsNugetPackageVersion>3.3.2-beta1.20562.1</RoslynDiagnosticsNugetPackageVersion>
-    <MicrosoftCodeAnalysisNetAnalyzersVersion>5.0.0-rc2.20453.1</MicrosoftCodeAnalysisNetAnalyzersVersion>
+    <MicrosoftCodeAnalysisNetAnalyzersVersion>5.0.3</MicrosoftCodeAnalysisNetAnalyzersVersion>
```

Comment:
Build is failing as the new NuGet package is not available in any of our consumed feeds, even though it is available on nuget.org. Seems that we do not list nuget.org as our feed, but instead use dotnet-public for this package: https://github.com/dotnet/roslyn/blob/a65055189039591201df2299db5b490025fad9b2/NuGet.config#L10

https://dev.azure.com/dnceng/public/_packaging?_a=package&feed=dotnet-public&package=Microsoft.CodeAnalysis.NetAnalyzers&protocolType=NuGet

Couple of questions for @dotnet/roslyn-infrastructure @mmitche:
1) https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers/5.0.3 is available. How do we get the same package published on dotnet-public feed? Or does this happen automatically after some time of the package getting published on nuget.org?
2) Is there a specific reason why we do not use nuget.org as a feed? If not, I'd prefer just to do so in this PR and avoid the same issue in future.

Reply:
I am going to unblock the PR for now by moving to an even newer 6.0.x preview version from dotnet6 feed.